command/agent/config.go

```diff
@@ -1302,6 +1306,16 @@ func (a *ClientConfig) Merge(b *ClientConfig) *ClientConfig {
 		result.ServerJoin = result.ServerJoin.Merge(b.ServerJoin)
 	}
 
+	if len(a.HostVolumes) == 0 && len(b.HostVolumes) != 0 {
```

Comment:
This feels like an optimization that be implemented inside `structs.HostVolumeSliceMerge`?  Not obvious to me if if it's worth it considering that the iteration would just skip `a` if it's empty?

Also, noticed that this implementation differs from `HostVolumeSliceMerge` in that `HostVolumeSliceMerge` checks for name collisions even within a single slice

Reply:
Cleaned this up a little, but this is the same pattern we use elsewhere. A Merge function doesn't necessarily want this optimisation imo.

Reply:
`result.HostVolumes` don't get deep copied if `len(a.HostVolume) != 0 && len(b.HostVolume) == 0` unlike rest of cases; is that intentional?